### ISSUE TITLE
Drop field "clli" from object fac

### ISSUE THREAD

**Issue by arnoldnipper** [peeringdb/peeringdb#945]
Spawned from #895 as a separate issue. For previous discussion see also there

**Comment by martinhannigan** [comment]
Once again, I'll point out that BOTH issues are really about what peeringDB
wants to be when it grows up.

*What uses a CLLI code?*

Buildings
Telco MMR's
Telco transport switch ports
Telco ethernet switch ports
Telco router switch ports
Manholes
Handholes
In street regen, huts, etc.

*What does not use a CLLI code?*

IXPs


*How much does it cost for a CLLI code to use properly? *

There is a cost for a single CLLI
There is a cost for the lookup that translates to address and GEO/LAT LONG


*Support to remove means:*

Telcos can't use it to provision
IXPs can't use it to provision

*Support to keep means:*

Telcos can use it to provision and probably do, but not facing IXPs and
other non LEC networks

*Thoughts and questions*

ZERO value for IXPs
All value is concentrated for LEC's
How many LEC's are using CLLI via PeeringDB to provision: My guess is ZERO
One value add could be a mopping of CLLI building to GEO code (see other
issue) but it will be expensive and accuracy difference may not be worth it.

In the absence of data of who uses it, I support dropping it. If some
evidence surfaces of actual wide spread use and accuracy, I would change it.


Thanks

**Comment by arnoldnipper** [comment]
> In the absence of data of who uses it, I support dropping it. I

+100

**Comment by ynbrthr** [comment]
> If some evidence surfaces of actual wide spread use and accuracy, I would change it. Thanks

Same. This is where we are: before dropping it, do we users actually using that field, or who would be using that field should it be more accurate.

**Comment by martinhannigan** [comment]
On Tue, Mar 16, 2021 at 7:59 PM Yan Berthier ***@***.***>
wrote:

> If some evidence surfaces of actual wide spread use and accuracy, I would
> change it. Thanks
>
> Same. This is where we are: before dropping it, do we users actually using
> that field, or who would be using that field should it be more accurate.
>

Here is a lookup of a city/st short code:

https://www.telcodata.us/search-area-code-exchange-by-clli?cllicode=KYWSFLMADS0


Tell me what you see? :-)

PDB is not the source of truth for CLLI. If someone is using it their
referring back to an expensive database service mainly oriented to voice,
but has some outside plant characteristics.

How many PDB CLLI codes are anything more than localizing city and state
which is duplicative of City, St. and geo code. (I am, but I only did it
because the field was there and the wrong data was already entered into
it). No one ever asks me for it live or otherwise and I’ve never seen it in
an LOA.

Hope that helps,

-M<

**Comment by mustangthz** [comment]
I’m opposed to dropping the CLLI field.

**Comment by martinhannigan** [comment]
Fair enough. Just not certain why.

**Comment by mustangthz** [comment]
Because there has been zero justification to remove a searchable feature that exists today. On the previous GitHub issue we discussed NPA/NXX which clearly has less value with all the overlays. CLLI is still useful for any telco.

**Comment by arnoldnipper** [comment]
Adding @job and @mahtin because they also were on the original thread

**Comment by mahtin** [comment]
I'm in agreement with @martinhannigan (which is kinda rare; but happens). He said: `PDB is not the source of truth for CLLI` and I agree with that wholeheartedly. Either do CLLI's with full gusto, which I think is outside the scope of PDB, or drop them. 
```
$ curl https://www.peeringdb.com/api/fac | jq -r '.data[]|.country,.clli' | paste - - | wc -l
4091
$
```
Of those 4091 facilities in PDB, only 478 have CLLI codes (see https://pastebin.com/raw/eyBtz4iz). Yet only 309 are in the US, 128 in Canada, and 39 in Mexico. Just for reference, a total of 1092 facilities are in the US or around 1/4 of PDB. The CLLI codes are not very consistent - and could well be wrong in many cases (again, I point made by @martinhannigan).

Stick to the knitting - focus on the continuing improvements in the geo-database'ing side of PDB search.

**Comment by martinhannigan** [comment]
And to elaborate on code validity:

Length Entities Valid?
6 202 Yes
7 1 No
8 125 Yes
9 2 Yes



Valid 329
Geo Only 202



Total Possible Useful 127

**Comment by mustangthz** [comment]
@martinhannigan 6 is valid.  So, you have 531 valid CLIIs.  I agree that 202 only provides you the municipality. That's likely a good start for a telco guy to help locate things.  

@mahtin you need to refine your facilities count to countries that use CLLI's to get a more accurate percentage of facilities that could have a CLLI and do have a CLLI.  Using the global facilities count is useless.

**Comment by martinhannigan** [comment]
I excluded MX. The train already left the station IMHO. I also agree with
@mahtin but I usually also agree with @martinhannigan, unlike him.

However, what the six byte codes actually show to me (and as I experienced
TWICE) is that the AC likely entered it, not the user. Which means they're
even more invalid. Nothing that a decent referred geo code hit won't do
better in terms of city, state and proximity if that's the only argument to
keep them.

So I continue to be -1. This horse is deader than dead.
